### DISCUSSION TITLE
Proposal to Add Template Support to p-toggleButton Component in PrimeNG for Custom Icons

### DISCUSSION THREAD

**Discussion by JuamBer** [primefaces/community#601] [PrimeNG] [answered]
I am currently using the p-toggleButton component in PrimeNG for my Angular application. While the component works well, I've noticed that it lacks support for custom icon templates. This limitation forces us to import entire icon libraries like FontAwesome, even if we only need a few icons, which can negatively impact the performance of our applications.

Here is an example of how I'm currently using the component:
```
<p-toggleButton onIcon="fas fa-check" offIcon="fas fa-times"></p-toggleButton>
```

To improve this, I propose adding template support to the p-toggleButton component. This would allow us to use custom icons without having to import the entire icon library. For example:
```
<p-toggleButton>
  <ng-template pTemplate="onIcon">
    <fa-icon [icon]="faCheck"></fa-icon>
  </ng-template>
  <ng-template pTemplate="offIcon">
    <fa-icon [icon]="faTimes"></fa-icon>
  </ng-template>
</p-toggleButton>
```
This approach would give developers more flexibility and control over the icons used in the p-toggleButton component, and could potentially improve application performance by reducing the need to import large icon libraries.

I would appreciate any thoughts or feedback on this proposal.

**Comment by SoyDiego** [comment]
> I am currently using the p-toggleButton component in PrimeNG for my Angular application. While the component works well, I've noticed that it lacks support for custom icon templates. This limitation forces us to import entire icon libraries like FontAwesome, even if we only need a few icons, which can negatively impact the performance of our applications.
> 
> Here is an example of how I'm currently using the component:
> 
> ```
> <p-toggleButton onIcon="fas fa-check" offIcon="fas fa-times"></p-toggleButton>
> ```
> 
> To improve this, I propose adding template support to the p-toggleButton component. This would allow us to use custom icons without having to import the entire icon library. For example:
> 
> ```
> <p-toggleButton>
>   <ng-template pTemplate="onIcon">
>     <fa-icon [icon]="faCheck"></fa-icon>
>   </ng-template>
>   <ng-template pTemplate="offIcon">
>     <fa-icon [icon]="faTimes"></fa-icon>
>   </ng-template>
> </p-toggleButton>
> ```
> 
> This approach would give developers more flexibility and control over the icons used in the p-toggleButton component, and could potentially improve application performance by reducing the need to import large icon libraries.
> 
> I would appreciate any thoughts or feedback on this proposal.

Hi @JuamBer, I replied your issue https://github.com/primefaces/primeng/issues/14267 about toggle button.

**Reply by JuamBer** [reply]
Hi Diego!

The issue has been closed and apparently when activating strict typescript mode the error disappeared, a bit strange.

But I'm still interested in having the possibility of passing the states of the p-toggleButton as templates.

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Thanks for reporting the missing icon template. We've added it and it'll be available on next release. https://github.com/primefaces/primeng/issues/14286

**Reply by JuamBer** [reply]
Thank you so much!
